src/main/scala/apiimport/persistence/ManifestPersistor.scala

```diff
@@ -0,0 +1,130 @@
+          }
+          .collect {
+            case Some(tuple) => tuple
+          }
+    }
+    .mapAsync(12) {
+      case (zipFile, jsonFile, vm) => addDowWoy(zipFile, jsonFile, vm)
+    }
+    .mapAsync(6) {
+      case (zipFile, jsonFile, vm, dow, woy) =>
+        removeExisting(zipFile, jsonFile, vm, dow, woy)
+          .flatMap(_ => persistManifest(zipFile, jsonFile, vm, dow, woy))
+    }
+
+  val con: db.tables.profile.backend.DatabaseDef = db.con
+
+  def persistManifest(zf: String, jf: String, vm: VoyageManifest, dow: Int, woy: Int): Future[Int] = con
+    .run(manifestTable.rowsToInsert(vm, dow, woy, jf))
+    .flatMap { _ => persistProcessedJsonRecord(zf, jf, vm) }
+
+  def persistProcessedJsonRecord(zf: String, jf: String, vm: VoyageManifest): Future[Int] = {
+    val suspiciousDate: Boolean = scheduledIsSuspicious(zf, vm)
+    val processedAt = new Timestamp(SDate.now().millisSinceEpoch)
+    val processedJsonFileToInsert = db.tables.ProcessedJson += db.tables.ProcessedJsonRow(zf, jf, suspiciousDate, success = true, processedAt)
+    con.run(processedJsonFileToInsert)
+  }
+
+  def persistFailedJsonRecord(zf: String, jf: String): Future[Int] = {
+    val processedAt = new Timestamp(SDate.now().millisSinceEpoch)
+    val processedJsonFileToInsert = db.tables.ProcessedJson += db.tables.ProcessedJsonRow(zf, jf, suspicious_date = false, success = false, processedAt)
+    con.run(processedJsonFileToInsert)
+  }
+
+  def persistProcessedZipRecord(zf: String, success: Boolean): Future[Int] = {
+    val processedAt = new Timestamp(SDate.now().millisSinceEpoch)
+    val processedZipFileToInsert = db.tables.ProcessedZip += db.tables.ProcessedZipRow(zf, success, processedAt)
+    con.run(processedZipFileToInsert)
+  }
+
+  def scheduledIsSuspicious(zf: String, vm: VoyageManifest): Boolean = {
+    val maybeSuspiciousDate: Option[Boolean] = for {
+      zipDate <- zipFileDate(zf)
+      scdDate <- vm.scheduleArrivalDateTime
+    } yield {
+      scdDate.millisSinceEpoch - zipDate.millisSinceEpoch > 2 * oneDayMillis
+    }
+
+    maybeSuspiciousDate.getOrElse(false)
+  }
+
+  def removeExisting(zf: String, jf: String, vm: VoyageManifest, dow: Int, woy: Int): Future[(String, String, VoyageManifest, Int, Int)] = {
+    val schTs = new Timestamp(vm.scheduleArrivalDateTime.map(_.millisSinceEpoch).getOrElse(0L))
+    val value = db.tables.VoyageManifestPassengerInfo.filter(r => {
+      r.event_code === vm.EventCode &&
+        r.arrival_port_code === vm.ArrivalPortCode &&
+        r.departure_port_code === vm.DeparturePortCode &&
+        r.scheduled_date === schTs &&
+        r.voyage_number === vm.VoyageNumber.toInt
+    })
+
+    con.run(value.delete).map(deletedCount => {
+      if (deletedCount > 0) log.info(s"Removed $deletedCount existing entries")
+      (zf, jf, vm, dow, woy)
+    })
+  }
+
+  def addDowWoy(zipFile: String, jsonFile: String, vm: VoyageManifest): Future[(String, String, VoyageManifest, Int, Int)] = {
```

Comment:
DowWoy is probably not that obvious to a future person looking at this code, maybe a long name would be better here, addDayOfWeekAndWeekOfYear

Reply:
done